baseimage/Makefile

```diff
@@ -1,18 +1,36 @@
-NAME="hyperledger/fabric-baseimage"
+NAME=hyperledger/fabric-baseimage
+VERSION=0.0.9
+ARCH=$(shell uname -m)
+DOCKER_TAG ?= $(ARCH)-$(VERSION)
 VAGRANTIMAGE=packer_virtualbox-iso_virtualbox.box
 
+DOCKER_BASE_x86_64=ubuntu:trusty
+
+DOCKER_BASE=$(DOCKER_BASE_$(ARCH))
+
+ifeq ($(DOCKER_BASE), )
+$(error "Architecture \"$(ARCH)\" is unsupported")
+endif
+
+
 # strips off the post-processors that try to upload artifacts to the cloud
 packer-local.json: packer.json
 	jq 'del(."post-processors"[0][1]) | del(."post-processors"[1][1])' packer.json > $@
 
-all:
-	packer build packer.json
+all: vagrant docker
 
 $(VAGRANTIMAGE): packer-local.json
+	BASEIMAGE_RELEASE=$(VERSION) \
 	packer build -only virtualbox-iso packer-local.json
 
-docker: packer-local.json
-	packer build -only docker packer-local.json
+Dockerfile: Dockerfile.in
+	@echo "# Generated from Dockerfile.in.  DO NOT EDIT!" > $@
+	@cat Dockerfile.in | \
+	sed -e  "s|_DOCKER_BASE_|$(DOCKER_BASE)|" >> $@
+
+docker: Dockerfile Makefile
+	@echo "Generating docker"
+	@docker build -t $(NAME):$(DOCKER_TAG) .
```

Comment:
@ghaskins One thing that did bug me, is the use of @ in the makefile, though I think this docker build command is the only one that I would really remove it from.. Unless docker doesnt report failures via $? properly and should be wrapped in some sort of grep?

One could argue the same about @ cat since if it failed, dockerbuild would foul up too without a generated dockerfile to follow...